hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -632,19 +632,24 @@ public void run() {
             }
             partitionServiceLock.lock();
             try {
-                triggerRepartitioningWhenClusterStateAllowsMigration = !isMigrationAllowedByClusterState();
-                if (triggerRepartitioningWhenClusterStateAllowsMigration && logger.isFineEnabled()) {
-                    logger.fine("Migrations are not allowed yet, "
-                            + "repartitioning will be triggered when cluster state allows migrations.");
+                if (!node.getClusterService().getClusterState().isMigrationAllowed()) {
+                    triggerRepartitioningWhenClusterStateAllowsMigration = true;
```

Comment:
Do we need to reset this to `false` once we actually manage to trigger repartitioning on cluster state change?

Reply:
You're right, I'll fix that.